src/test/java/bio/terra/service/snapshot/SnapshotDaoTest.java

```diff
@@ -509,12 +535,34 @@ public void snapshotEnumerateTest() {
         filteredAllTagsMatchEnum.getTotal(),
         equalTo(6));
 
+    // Test filtering on DUOS IDs
+    MetadataEnumeration<SnapshotSummary> filteredDuosIdsMatchEnum =
+        snapshotDao.retrieveSnapshots(
+            0, 6, null, null, null, null, datasetIds, snapshotIds, null, List.of(duosId));
+    List<SnapshotSummary> filteredDuosIdsMatchSnapshots = filteredDuosIdsMatchEnum.getItems();
+    assertThat(
+        "snapshot filter by duosid returns correct filtered total",
+        filteredDuosIdsMatchSnapshots,
+        hasSize(2));
+    assertThat(
+        "snapshot filter by duosid returns correct snapshot",
+        filteredDuosIdsMatchSnapshots.stream().map(SnapshotSummary::getId).sorted().toList(),
+        equalTo(snapshotIds.subList(0, 2).stream().sorted().toList()));
```

Comment:
I think `containsInAnyOrder()` lets you avoid sorting here
```suggestion
        filteredDuosIdsMatchSnapshots.stream().map(SnapshotSummary::getId).toList(),
        containsInAnyOrder(snapshotIds.subList(0, 2)));
```

Reply:
ah not quite.  This is why I would love to (some day) switch to AssertJ...you need to convert to an array.  containsInAnyOrder looks at the list as a single element as opposed to it's members :(

Reply:
That's true, although you could use `containsInAnyOrder(snapshotIds.get(0), snapshotIds.get(1))`. Even so I think this is better than the current code as you don't need to sort, and you don't need to check `hasSize()` since `containsInAnyOrder()` does that for you.

IMO in general checking both `hasSize()` and the contents of an array seems like a case where the test is probably redundant and could be simplified.

Also, below in `verifySnapshotProject()` there are many different assertions on the same value. That seems like another redundant case where we could use `allOf()` to combine these checks.

```
  private void verifySnapshotProject(Snapshot snapshot) {
    SnapshotProject snapshotProject = snapshotDao.retrieveSnapshotProject(snapshot.getId());
    assertThat(
        snapshotProject,
        allOf(
            hasProperty("id", is(snapshot.getId())),
            hasProperty("name", is(snapshot.getName())),
            hasProperty("profileId", is(snapshot.getProfileId())),
            hasProperty("dataProject", is(snapshot.getProjectResource().getGoogleProjectId()))));
    Dataset sourceDataset = snapshot.getSourceDataset();
    var projectResource = sourceDataset.getProjectResource();
    assertThat(
        snapshotProject.getSourceDatasetProjects(),
        containsInAnyOrder(
            allOf(
                hasProperty("id", is(sourceDataset.getId())),
                hasProperty("name", is(sourceDataset.getName())),
                hasProperty("profileId", is(projectResource.getProfileId())),
                hasProperty("dataProject", is(projectResource.getGoogleProjectId())))));
  }
```